packages/cmrjs/src/cmr-utils.js

```diff
@@ -1098,6 +1099,11 @@ async function getGranuleTemporalInfo(granule) {
 
   const cmrFilename = getS3UrlOfFile(cmrFile[0]);
 
+  if (!(await s3ObjectExists(parseS3Uri(cmrFilename)))) {
```

Comment:
Taking a closer look at `s3ObjectExists` do we need to await `headObject` in that function https://github.com/nasa/cumulus/blob/master/packages/aws-client/src/S3.ts#L191-L197 ? Can we add unit test to https://github.com/nasa/cumulus/blob/master/packages/aws-client/tests/test-S3.js?